hooks/shellcheck.sh

```diff
@@ -7,44 +7,62 @@ set -e
 # workaround to allow GitHub Desktop to work, add this (hopefully harmless) setting here.
 export PATH=$PATH:/usr/local/bin
 
-exit_status=0
-enable_list=""
-
-parse_arguments() {
-	while (($# > 0)); do
-		# Grab param and value splitting on " " or "=" with parameter expansion
-		local PARAMETER="${1%[ =]*}"
-		local VALUE="${1#*[ =]}"
-		if [[ "$PARAMETER" == "$VALUE" ]]; then VALUE="$2"; fi
-		shift
-		case "$PARAMETER" in
-		--enable)
-			enable_list="$enable_list $VALUE"
-			;;
-		-*)
-			echo "Error: Unknown option: $PARAMETER" >&2
-			exit 1
-			;;
-		*)
-			files="$files $PARAMETER"
-			;;
-		esac
-	done
-	enable_list="${enable_list## }" # remove preceeding space
+readonly SHEBANG_REGEX='^#!\(/\|/.*/\|/.* \)\(\(ba\|da\|k\|a\)*sh\|bats\)$'
+
+function shellcheck_files {
+  local -r enable_list="$1"
+  local -r files="$2"
+
+  local exit_status=0
+
+  for file in $files; do
```

Comment:
Should `files` be declared as `local -a`? Does it need array parsing (e.g., ,`("$2")`)?

Reply:
This is intentionally constructing a space separated string and passing that around to avoid the mess of passing arrays around in bash. This works because `for` loop on string is inferred to be word iteration.